src/lib/components/Modal/ModalTitle.jsx

```diff
@@ -0,0 +1,45 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  withGlobalProps,
+} from '../../provider';
+import styles from './ModalTitle.scss';
+
+export const ModalTitle = ({
```

Comment:
I'm unsure about this component.

Would it be possible to not have it and pass in a `node` to have it more open?

Specifically I'm interested in this situation:
```
<Icon /> <h2>My Title</h2>                                                     <ModalCloseButton />
```
I think the current solution does not make this possible. If `node` could be passed in, we could simply do:

```jsx
<ModalHeader>
  <span>
    <Icon />
    <h2>My Title</h2>
  </span>
  <ModalCloseButton />
</ModalHeader>
```

Reply:
Note: This has been implemented by @adamkudrna in PR that was merged into this PR. Not sure how to handle it as it was already approved. In such case, @adamkudrna as author, should change it here.

@mbohal Originally, I implemented it like you, but it is more complicated than you think. Adam would describe it better, but due to some CSS things (specificity or selectors or something like that), we would not be able to obtain correct appearance of headings and the only option is dirty option with `!important` and disabling stylelint. Correct way it own component which internally solves styles. I was not able to do it, so Adam came up with `ModalTitle`. And I think, it is right way. Another libraries implement it same way, maybe because of same problems.

In case you want advanced layout, use custom layout with custom components. But I guarantee you that 98% people will not need it. Personally, I can't remember situation when anything else than text was used as title of modal - if we speak about serious web apps.

Reply:
> it is more complicated than you think

Yah, I though so…

> In case you want advanced layout, use custom layout with custom components

Sure, that would work.

Another solution might be for the `ModalTitle` not to render the `<hx>`, but just a `<span>`. The user would then pass into it the contents, typically `<h2>My title<h2>`. This would also remove the need for the `level` prop. The downsides are:

* allows users to not use the heading (`<hx>`) tags in modal titles
* in the typical use case with `<h2>` it is slightly more verbose

Reply:
I've carefully considered what @mbohal wrote and I still think this is the right solution.

1. We need `ModalTitle` to be able to style the headline.
2. We want to output the best possible markup out of the box, without making developers think about the problematics of modal headlines. Those who need it can easily adjust heading level or replace `ModalTitle` with a custom component.
   <img width="454" alt="obrazek" src="https://user-images.githubusercontent.com/5614085/182191473-80b7a7f2-8fe4-4207-bdad-60f9255ca2c3.png">
3. Placing an icon along the headline is already possible (though it requires some styling):
   <img width="717" alt="Snímek obrazovky 2022-08-01 v 17 57 20" src="https://user-images.githubusercontent.com/5614085/182191298-15995865-aea5-4ef5-81f6-bfd27820fd1e.png">

Reply:
If @mbohal is ok with it, please mark this conversation as resolved.

Reply:
@bedrich-schindler @adamkudrna I'll trust your judgment on this.